task/sast-coverity-check-oci-ta/0.3/sast-coverity-check-oci-ta.yaml

```diff
@@ -874,13 +883,35 @@ spec:
           update-ca-trust
         fi
 
+        # fallback to buildless scan if we have no scan results from buildful
         if [ -z "$(ls /shared/sast-results/)" ]; then (
           set +e
           set -x
 
-          # fallback to buildless scan if we have no scan results from buildful
-          # shellcheck disable=SC2086
-          env HOME=/var/tmp/coverity/home /opt/coverity/bin/coverity capture --disable-build-command-inference --dir /tmp/idir --project-dir "/var/workdir"
+          # generate full path for each directory in TARGET_DIRS separate by comma
+          SOURCE_CODE_DIR="/var/workdir/source"
+          declare -a ALL_TARGETS
+          IFS=","
+          for d in $TARGET_DIRS; do
+            potential_path="${SOURCE_CODE_DIR}/${d}"
+            resolved_path=$(realpath -m "$potential_path")
+
+            # ensure resolved path is still within SOURCE_CODE_DIR
+            if [[ ! "$resolved_path" == "$SOURCE_CODE_DIR"* ]]; then
+              echo "Error: path traversal attempt, '$potential_path' is outside '$SOURCE_CODE_DIR'"
+              exit 1
+            fi
+
+            # ensure directory exists, as `coverity capture` doesn't
+            [ -d "$resolved_path" ] || (echo "Directory $resolved_path does not exist" && exit 1)
+
+            ALL_TARGETS+=("$resolved_path")
+          done
+
+          for d in "${ALL_TARGETS[@]}"; do
+            # shellcheck disable=SC2086
+            env HOME=/var/tmp/coverity/home /opt/coverity/bin/coverity capture --disable-build-command-inference --dir /tmp/idir --project-dir "${d}"
```

Comment:
I think a safer approach would be to include the whole chain from `coverity capture` to `cov-format-errors` in the loop body and merge the resulting JSON files as we do it with the buildful capture.  This is what Coverity Support replied on Case 01188375:
> In particular, the Coverity CLI does not purport to support a workflow where capture is invoked multiple times across multiple project directories.

Reply:
Ah, good info! Done.